pkg/kubectl/resize.go

```diff
@@ -33,23 +34,45 @@ type ResizePrecondition struct {
 	ResourceVersion string
 }
 
+// A PreconditionError is returned when a replication controller fails to match
+// the resize preconditions passed to kubectl.
 type PreconditionError struct {
 	Precondition  string
 	ExpectedValue string
 	ActualValue   string
 }
 
-func (pe *PreconditionError) Error() string {
+func (pe PreconditionError) Error() string {
 	return fmt.Sprintf("Expected %s to be %s, was %s", pe.Precondition, pe.ExpectedValue, pe.ActualValue)
 }
 
+// Types of ControllerResizeErrors
+const (
```

Comment:
This is one of the things go's type system is good at.

```go

type ControllerResizeErrorType int

const (
	ControllerResizeGetFailure ControllerResizeErrorType = iota
	ControllerResizeUpdateFailure
)
```

Unfortunately this package has too much stuff in it and therefore we probably should make the name long enough to not collide.